### PR TITLE
feat: implement api to add notifications to db

### PR DESCRIPTION
**Pull request by LakshayaT**
This PR implements the feature of implementing a POST API for adding new notifications to the database.

Related changes:
- Add notification controller to accept POST request for add
- Add notification service to execute code to create notification 
- Create required notification module, register entity 
- Update database configuration to autoload entities
- Add a constants/notifications.ts file to store constant values

Method: `POST`
Endpoint: `/notifications`
Sample Request Body:
```json
{
  "channelType": "smtp-server",
  "data": {
    "from": "sender@example.com",
    "to": "recipient@example.com",
    "subject": "Test subject",
    "text": "This is a test notification",
    "html": "<b>This is a test notification</b>"
  },
  "createdBy": ""
}
```
Response:
```json
{
  "notification": {
    "channelType": "smtp-server",
    "data": {
      "from": "sender@example.com",
      "to": "recipient@example.com",
      "subject": "Test subject",
      "text": "This is a test notification",
      "html": "<b>This is a test notification</b>"
    },
    "deliveryStatus": 1,
    "createdBy": "",
    "createdOn": "2023-08-25T10:55:36.794Z",
    "result": null,
    "modifiedBy": null,
    "modifiedOn": null,
    "id": 1
  }
}
```

### CODE REVIEW

File: src/models/notifications/notifications.controller.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Controller, Post, Body } from '@nestjs/common';
+import { NotificationsService } from './notifications.service';
+import { NOTIFICATION_CREATE_SUCCESS_MSG } from 'src/common/constants/notifications';
+
+@Controller('notifications')
+export class NotificationsController {
+  constructor(private readonly notificationService: NotificationsService) {}
+
+  @Post('add')
```
Comment (osm-vishnukyatannawar) [inline src/models/notifications/notifications.controller.ts:9]: end point should be just

/notifications

POST

File: src/models/notifications/notifications.controller.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  async addNotification(@Body() notificationData: any) {
+    const createdNotification =
+      await this.notificationService.createNotification(notificationData);
+    return {
+      message: NOTIFICATION_CREATE_SUCCESS_MSG,
+      notification: createdNotification,
```
Comment (osm-vishnukyatannawar) [inline src/models/notifications/notifications.controller.ts:15]: Just return this. The message is not required.

**Review by osm-vishnukyatannawar — APPROVED** [review]

File: src/models/notifications/notifications.controller.ts (hunk 1 above)
Comment (LakshayaT) [reply src/models/notifications/notifications.controller.ts]: API end point has been updated from `/notifications/add` to `/notifications` in latest commit

File: src/models/notifications/notifications.controller.ts (hunk 2 above)
Comment (LakshayaT) [reply src/models/notifications/notifications.controller.ts:13]: The message value, along with the constant storing it has been removed in the latest commit.

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**